libs/NfcCoreLib/lib/Nci/phNciNfc_NfceeMgmt.c

```diff
@@ -161,6 +161,10 @@ static NFCSTATUS phNciNfc_CompleteModeSetSequence(void *pContext, NFCSTATUS wSta
              (phNciNfc_IsVersion2x(pNciCtx) &&
                 wStatus != NFCSTATUS_SUCCESS))
         {
+            pNciCtx->IfNtf = pNciCtx->IfModeSetNtf;
+            pNciCtx->IfNtfCtx = pNciCtx->IfNtfModeSetCtx;
```

Comment:
By doing this you're assuming that `pNciCtx->IfNtf` and `pNciCtx->IfNtfCtx` are both `NULL`. Why is this assumption correct?
If it's not always the case then previous values of these vars will be lost.

Reply:
The callback data(function and context) intended, is saved before at line 652  before sending the command and same data is copied back on the response.Done on intent for handling NCI 1.0 and NCI 2.0 mode set behaviour.
There is no loss of values.

Reply:
I think we might have a misunderstanding here.
Before this assignment on line 164: `pNciCtx->IfNtf = pNciCtx->IfModeSetNtf;` there could be another callback stored in `pNciCtx->IfNtf`. By assigning MODE_SET_NTF callback, the old value of `pNciCtx->IfNtf` is getting lost.

Reply:
pNciCtx->IfNtf  will have valid data till the response or ntf comes to MW.
Once the call back is called no data present.
As the data is copied before sending the command.
And same data is copied before calling the callback once response received.
Overwriting is done on purpose and there is no loss of old data.